src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/Decoder.cs

```diff
@@ -10,9 +10,9 @@ namespace System.IO.Compression
 {
     internal sealed class Decoder
     {
-        internal IntPtr State = IntPtr.Zero;
-        internal BrotliDecoderResult LastDecoderResult = BrotliDecoderResult.NeedsMoreInput;
-        internal MemoryStream BufferStream;
+        internal IntPtr _state = IntPtr.Zero;
```

Comment:
We went away from the problem by changing "state" to an internal property with a private setter. I think reaching into a class and touching private fields is the wrong approach here.